src/amo/components/GetFirefoxButton/index.js

```diff
@@ -84,14 +113,41 @@ export const GetFirefoxButtonBase = (
         addon,
         `addon is required for buttonType ${GET_FIREFOX_BUTTON_TYPE_ADDON}`,
       );
-      buttonText = i18n.gettext('Only with Firefox—Get Firefox Now');
+
+      const promotedCategory = _getPromotedCategory({
+        addon,
+        clientApp,
+        forBadging: true,
+      });
+
+      const supportsRTAMO =
+        promotedCategory === RECOMMENDED && clientApp === CLIENT_APP_FIREFOX;
+
+      if (!useNewVersion) {
+        buttonText = i18n.gettext('Only with Firefox—Get Firefox Now');
+      } else {
+        const downloadTextForRTAMO =
+          addon.type === ADDON_TYPE_STATIC_THEME
+            ? i18n.gettext('Download Firefox and get the theme')
+            : i18n.gettext('Download Firefox and get the extension');
+        buttonText = supportsRTAMO
+          ? downloadTextForRTAMO
+          : i18n.gettext('Download Firefox');
+        calloutText =
+          addon.type === ADDON_TYPE_STATIC_THEME
+            ? i18n.gettext(`You'll need Firefox to use this theme`)
+            : i18n.gettext(`You'll need Firefox to use this extension`);
+      }
       puffy = true;
+      utmCampaign = `amo-fx-cta-${addon.id}-${variant}`;
```

Comment:
mmm, tbh I'd prefer that these `utmCampaign` values were defined in a single place, or built in a utility function.  But for an experiment it's fine...?

Reply:
This is a good idea. I'll do this.